clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -441,4 +500,25 @@ private Result checkDistributors(Node node, int clusterStateVersion) {
         return allowSettingOfWantedState();
     }
 
+    private Collection<Group> groupsWithStorageNodesInMaintenance() {
+        var groupIndexes = clusterInfo.getStorageNodeInfos().stream()
+                                      .filter(sni -> MAINTENANCE.equals(sni.getWantedState().getState()))
+                                      .map(NodeInfo::getGroup)
+                                      .filter(Objects::nonNull)
+                                      .filter(Group::isLeafGroup)
+                                      .map(Group::getIndex)
```

Comment:
re: offline discussion about the many fun and exciting quirks of groups, consider `Group::getPath` to disambiguate nested groups with the same logical index